src/main/java/org/sagebionetworks/web/client/widget/asynch/IsACTMemberAsyncHandlerImpl.java

```diff
@@ -46,35 +54,57 @@ public void invoke(Boolean isACTMember) {
 		});
 	}
 
+	@Override
+	public FluentFuture<Boolean> isACTActionAvailable() {
+		return isACTMember().transform(isActMember -> isActMember && visible, directExecutor());
+	}
+
 	@Override
 	public void isACTMember(final CallbackP<Boolean> callback) {
+		this.isACTMember().addCallback(new FutureCallback<Boolean>() {
+			@Override
+			public void onSuccess(@NullableDecl Boolean result) {
+				callback.invoke(result);
+			}
+
+			@Override
+			public void onFailure(Throwable caught) {
+				// log the error, and tell client that user is not part of the ACT (give the developer something to
+				// look at if something goes wrong for an ACT member)
+				jsniUtils.consoleError(caught.getMessage());
+				callback.invoke(false);
+
+			}
```

Comment:
Same failure scenario as below. As of now, this should never happen because it calls a Future that can never fail.